x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/PreAnalyzer.java

```diff
@@ -45,19 +48,20 @@ public PreAnalysis preAnalyze(LogicalPlan plan) {
     }
 
     protected PreAnalysis doPreAnalyze(LogicalPlan plan) {
-        List<TableInfo> indices = new ArrayList<>();
+        Set<TableInfo> indices = new HashSet<>();
         List<Enrich> unresolvedEnriches = new ArrayList<>();
         List<TableInfo> lookupIndices = new ArrayList<>();
 
         plan.forEachUp(UnresolvedRelation.class, p -> {
-            List<TableInfo> list = p.indexMode() == IndexMode.LOOKUP ? lookupIndices : indices;
-            list.add(new TableInfo(p.indexPattern()));
+            Collection<TableInfo> collection = p.indexMode() == IndexMode.LOOKUP ? lookupIndices : indices;
```

Comment:
I needed to make this change and the ones in `TableInfo` because otherwise queries using FORK would fail with the following errror:

https://github.com/elastic/elasticsearch/blob/d30296229bb095a38a6af93d9b8b8c8cab4156f1/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java#L434-L438

Note that this only ensures that we collect distinct values for `indices`, not `lookupIndices`, meaning there should be no effect on the lookup join functionality.

Reply:
This makes sense to. I don't remember exactly, but I did run into this in the past but then didn't need it for a separate reason.